src/actions/redeemParticipationBidV3.ts

```diff
@@ -0,0 +1,206 @@
+
+  txInitBatch.addSigner(mint);
+  txInitBatch.addTransaction(createMintTx);
+  txInitBatch.addTransaction(createAssociatedTokenAccountTx);
+  txInitBatch.addTransaction(mintToTx);
+
+  const newMint = mint.publicKey;
+  const newMetadataPDA = await Metadata.getPDA(newMint);
+  const newEditionPDA = await Edition.getPDA(newMint);
+
+  const metadataPDA = await Metadata.getPDA(originalMint);
+  const masterEditionPDA = await MasterEdition.getPDA(originalMint);
+  const masterEdition = await MasterEdition.load(connection, masterEditionPDA);
+
+  const prizeTrackingTicketPDA = await PrizeTrackingTicket.getPDA(auctionManagerPDA, originalMint);
+  const winIndex = bidState.getWinnerIndex(bidder.toBase58());
+
+  const desiredEdition = masterEdition.data.supply.add(new BN(1));
+  const editionMarkerPDA = await EditionMarker.getPDA(originalMint, desiredEdition);
+
+  let tokenPaymentAccount: PublicKey;
+  if (auctionTokenMint === NATIVE_MINT.toBase58()) {
+    const { account, createTokenAccountTx, closeTokenAccountTx } = await createWrappedAccountTxs(
+      connection,
+      bidder,
+      fixedPrice.toNumber(),
+    );
+    tokenPaymentAccount = account.publicKey;
+    txInitBatch.addTransaction(createTokenAccountTx);
+    txInitBatch.addSigner(account);
+    txMainBatch.addAfterTransaction(closeTokenAccountTx);
+  } else {
+    // TODO: find out what will happen if currency is not WSOL
+    tokenPaymentAccount = await Token.getAssociatedTokenAddress(
+      ASSOCIATED_TOKEN_PROGRAM_ID,
+      TOKEN_PROGRAM_ID,
+      new PublicKey(auctionTokenMint),
+      bidder,
+    );
+  }
+
+  const { authority, createApproveTx, createRevokeTx } = createApproveTxs({
+    account: tokenPaymentAccount,
+    owner: bidder,
+    amount: fixedPrice.toNumber(),
+  });
+  txMainBatch.addTransaction(createApproveTx);
+  txMainBatch.addAfterTransaction(createRevokeTx);
+  txMainBatch.addSigner(authority);
+
+  // create redeem bid
+  const redeemParticipationBidV3Tx = new RedeemParticipationBidV3(
+    { feePayer: bidder },
+    {
+      store,
+      vault: vault.pubkey,
+      auction,
+      auctionManager: auctionManagerPDA,
+      bidRedemption: bidRedemptionPDA,
+      bidMetadata: bidderMetaPDA,
+      safetyDepositTokenStore,
+      destination: recipient,
+      safetyDeposit: safetyDepositBox.pubkey,
+      bidder,
+      safetyDepositConfig: safetyDepositConfigPDA,
+      auctionExtended: auctionExtendedPDA,
+      newMint,
+      newEdition: newEditionPDA,
+      newMetadata: newMetadataPDA,
+      metadata: metadataPDA,
+      masterEdition: masterEditionPDA,
+      editionMark: editionMarkerPDA,
+      prizeTrackingTicket: prizeTrackingTicketPDA,
+      winIndex: winIndex !== null ? new BN(winIndex) : null,
+      transferAuthority: authority.publicKey,
+      tokenPaymentAccount,
+      acceptPaymentAccount,
+    },
+  );
+  txMainBatch.addTransaction(redeemParticipationBidV3Tx);
+  ////
```

Comment:
ditto above. If these are meant to e separators for easy parsing of this large file we should use more descriptive headers like:

```js
// -----------------
//  updatePrimarySaleHappenedViaTokenTx
// -----------------
```